crates/net/network/src/session/active.rs

```diff
@@ -563,6 +563,11 @@ impl Future for ActiveSession {
                         // notify the manager
                         return this.close_on_error(err, cx)
                     }
+                    
+                    if let Poll::Ready(Err(err)) = this.conn.poll_flush_unpin(cx) {
+                        debug!(target: "net::session", %err, remote_peer_id=?this.remote_peer_id, "failed to flush connection");
+                        return this.close_on_error(err, cx)
```

Comment:
this is slightly problematic because we only enter this if there's a new outgoing message

because the underlying sink has an internal buffer we always need to call this and should move this out of the while loop so that we first
* buffer as many messages as available
* always flush